docs/snapshots.md

```diff
@@ -0,0 +1,54 @@
+# Index Snapshots
```

Comment:
This is weird, this is shown as an addition for me, but didn't we have this doc?

Reply:
It was deleted along with the Rosetta stuff by mistake and readded here IIRC, my bad :(